client/src/hooks/useGetAd.js

```diff
@@ -26,20 +30,23 @@ export const useGetAd = () => {
     };
 
     const initAd = async () => {
-      try {
-        const res = await getAd(id);
-        await getMapData(res.zipCode);
+      if (loading) {
+        dispatch({ type: ACTIONS.LOAD_AD });
+      }
+      if (error) {
+        dispatch({ type: ACTIONS.ERROR, payload: { error: error } });
+      }
+      if (data) {
+        await getMapData(data.ad.zipCode);
         dispatch({
           type: ACTIONS.GET_AD,
-          payload: { ad: res },
+          payload: { ad: data.ad },
         });
-      } catch (error) {
-        dispatch({ type: ACTIONS.ERROR, payload: { error: error } });
       }
     };
 
     initAd();
-  }, [id]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
The comment here is intentional. It gets rid of the dependency warning in the console. it's complaining that we aren't passing dispatch in the dependency array. In this case, we do not want the page to re-render on every dispatch.